src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheck.java

```diff
@@ -800,6 +801,7 @@ private static boolean isAnonymousInnerClassEnd(int currentType, char nextChar)
     private boolean isEmptyBlock(DetailAST ast, int parentType) {
         return isEmptyMethodBlock(ast, parentType)
                 || isEmptyCtorBlock(ast, parentType)
+                && !isEmptyCtorBlock(ast)
```

Comment:
`isEmptyCtorBlock  && !isEmptyCtorBlock` looks impossible. Please merge this two methods into one.

Reply:
`isEmptyCtorBlock && !isEmptyCtorBlock` cannot be merged into one as-
```
public class foo {
    public foo() {}
}
```
`isNotRelevantSituation` method has to be false for the tokens to be examined. If we merge them into one then we would have to remove it from `isNotRelevantSituation` and would have to add conditions in `shouldCheckSeparationFromPreviousToken` and `shouldCheckSeparationFromNextToken`. Moreover while writing these conditions in `shouldCheckSeparation...` methods we cannot use
`isEmptyCtorBlock(DetailAST ast, int parentType)` as we have to examine `SLIST` and `RCURLY` tokens separately. 

I agree that in the current solution `isEmptyCtorBlock(DetailAST ast, int parentType)` is returning true whereas `!isEmptyCtorBlock(DetailAST ast)` is returning false for the same condition. 

**What can be done-**

- `isEmptyCtorBlock(DetailAST ast, int parentType)` can be completely removed. 
- A new method `isEmptyCtorBlockCheckedFromRcurly(DetailAST ast)` can be added. 
- `isEmptyCtorBlock(DetailAST ast)` can be renamed to `isEmptyCtorBlockCheckedFromSlist(DetailAST ast)`
- Finally instead of `isEmptyCtorBlock && !isEmptyCtorBlock`, 
```!isEmptyCtorBlockCheckedFromSlist && isEmptyCtorBlockCheckedFromRcurly``` can be added.

**What I suggest-**
As these methods can't be merged into one, separating them will be good but won't result in a DRY code, 
`isEmptyBlock(DetailAST ast, int parentType, int match)` already checks for `RCURLY` token.

While `isEmptyCtorBlock && !isEmptyCtorBlock` always returns false for `SLIST` token but conditions have been added to 
`shouldCheckSeparationFromNextToken` to prevent any bad behaviour.

Rest I'll proceed as you say.

Reply:
Thank you for the detailed explanation.

>     * Finally instead of `isEmptyCtorBlock && !isEmptyCtorBlock`,
>       `!isEmptyCtorBlockCheckedFromSlist && isEmptyCtorBlockCheckedFromRcurly` can be added.

This option will be most clear for the next ones who will change the code in the future.

Reply:
Done.